src/App.js

```diff
@@ -1,289 +1,182 @@
-      e.preventDefault(); // prevents refreshing
-      try {
-        const body = searchObject ;
-        const response = await fetch("http://localhost:5000/lastname", {
-          method: "POST", // fetch makes GET request by default
-          headers: { "Content-Type": "application/json" },
-          body: JSON.stringify(body)
-        });
-        const jsonData = await response.json();
-        setContacts(jsonData);
-        console.log(jsonData);
-        
-        console.log(response);
-        console.log(searchObject);
-        setStuff(jsonData);
-        // window.location = "/"; // refresh after done sending
-      } catch (error) {
-        console.error(error.message);  
-      }
+    try {
+      setContacts(await (await fetch(url)).json());
+    } catch (error) {
+      console.error(error.message);
     }
-  
-  const [stuff, setStuff] = useState([]);
+  };
 
-  return(
+  return (
     <>
-    <h2>{props.title}</h2>
-    <div>
-      <label>{props.title}
-        <input 
-          type="text"
-          placeholder="search by last name"
-          onChange={ (e) => setSearch(e.target.value) }
-        />
-      </label>
-
-
-      <button
-      onClick = { onSubmitSearch }
+      <h2>{props.title}</h2>
+      <form
+        onSubmit={(e) => {
+          e.preventDefault();
+          queryContacts(searchTerm)();
+        }}
       >
-        {props.title}
-      </button>
-
-      <button
-      onClick = { onSubmit0Search }
-      >
-        View All
-      </button>
-    </div>
-
-    <h2>{ stuff.length > 0 ? "" : "No" } {props.title} Results</h2>
-      { stuff.length > 0 ? <table>
-    <thead>
-      <tr>
-        <th>id</th>
-        <th>first name</th>
-        <th>last name</th>
-        <th>phone</th>
-        <th>email</th>
-      </tr>
-    </thead>
-    <tbody>
-    { contacts.map( contact => (
-      <tr>
-        <td>{contact.id}</td>
-        <td>{contact.first_name}</td>
-        <td>{contact.last_name}</td>
-        <td>{contact.phone_number}</td>
-        <td>{contact.email}</td>
-      </tr>
-    ) )
-    }
-    </tbody>
-  </table> : <p> No results found </p> }
+        <label>
+          {props.title}
+          <input
+            type="text"
+            placeholder="search by last name"
+            onChange={(e) => setSearchTerm(e.currentTarget.value)}
+          />
+        </label>
+
+        <button>{props.title}</button>
+        <button type="button" onClick={queryContacts()}>
+          View All
+        </button>
+      </form>
+
+      <h2>
+        {contacts.length > 0 ? "" : "No"} {props.title} Results
+      </h2>
+      {contacts.length > 0 ? (
+        <table>
+          <thead>
+            <tr>
+              <th>id</th>
+              <th>first name</th>
+              <th>last name</th>
+              <th>phone</th>
+              <th>email</th>
+            </tr>
+          </thead>
+          <tbody>
+            {contacts.map((contact) => (
+              <tr key={contact.id}>
```

Comment:
Don't forget your `key` prop on list items.

Reply:
I watched a lynda video segment about it last week and forgot how to do it. Thanks for showing how to implement it!